src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -6996,6 +6998,28 @@ static void sofia_handle_sip_r_options(switch_core_session_t *session, int statu
 	}
 }
 
+/* Check if we should skip directory lookup for 302 redirects */
+static switch_bool_t sofia_should_skip_directory_lookup(sofia_profile_t *profile, int status, const char *host)
+{
+	char domains[1024], *list[64];
+	int i;
+	
+	if (status != 302 || zstr(profile->redirect_no_lookup_domains) || zstr(host))
+		return SWITCH_FALSE;
+	
+	switch_copy_string(domains, profile->redirect_no_lookup_domains, sizeof(domains));
+	
+	for (i = 0; i < switch_separate_string(domains, ',', list, 64); i++) {
```

Comment:
switch_separate_string should not be in the condition in the for loop otherwise it always gonna perform tokenizing. Store the size of the separated list instead